optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/statistic/memoryuse/MemoryUseSubSingleStatistic.java

```diff
@@ -45,32 +45,39 @@ public MemoryUseSubSingleStatistic(SubSingleBenchmarkResult subSingleBenchmarkRe
                     + ") must be bigger than 0.");
         }
         this.timeMillisThresholdInterval = timeMillisThresholdInterval;
-        listener = new MemoryUseSubSingleStatisticListener();
     }
 
     // ************************************************************************
     // Lifecycle methods
     // ************************************************************************
 
     @Override
-    public void open(Solver<Solution_> solver) {
-        ((AbstractSolver<Solution_>) solver).addPhaseLifecycleListener(listener);
+    public void open(StatisticRegistry registry, Tags runTag, Solver<Solution_> solver) {
+        registry.addListener(SolverMetric.MEMORY_USE, new MemoryUseSubSingleStatisticListener(registry, runTag));
     }
 
     @Override
-    public void close(Solver<Solution_> solver) {
-        ((AbstractSolver<Solution_>) solver).removePhaseLifecycleListener(listener);
+    public void close(StatisticRegistry registry, Tags runTag, Solver<Solution_> solver) {
     }
 
-    private class MemoryUseSubSingleStatisticListener extends PhaseLifecycleListenerAdapter<Solution_> {
+    private class MemoryUseSubSingleStatisticListener implements Consumer<Long> {
 
         private long nextTimeMillisThreshold = timeMillisThresholdInterval;
+        private StatisticRegistry registry;
+        private Tags tags;
+
+        public MemoryUseSubSingleStatisticListener(StatisticRegistry registry, Tags tags) {
+            this.registry = registry;
+            this.tags = tags;
+        }
 
         @Override
-        public void stepEnded(AbstractStepScope<Solution_> stepScope) {
-            long timeMillisSpent = stepScope.getPhaseScope().calculateSolverTimeMillisSpentUpToNow();
+        public void accept(Long timeMillisSpent) {
             if (timeMillisSpent >= nextTimeMillisThreshold) {
-                pointList.add(new MemoryUseStatisticPoint(timeMillisSpent, MemoryUseMeasurement.create()));
+                registry.getGaugeValue(SolverMetric.MEMORY_USE, tags, memoryUse -> {
+                    pointList.add(new MemoryUseStatisticPoint(timeMillisSpent, new MemoryUseMeasurement(memoryUse.longValue(),
+                            (long) registry.find("jvm.memory.max").tags(tags).gauge().value())));
```

Comment:
Is `max` what we want? I'd expect we want the current usage, not the most it's even been.

Reply:
MemoryUseStatisticPoint use both current and max